metamist/parser/generic_metadata_parser.py

```diff
@@ -465,6 +468,16 @@ def collapse_arbitrary_meta(key_map: dict[str, str], row: GroupedRow):
         if not key_map or not row:
             return {}
 
+        def string_to_bool(value: Any) -> Any:
```

Comment:
You could just make this "guess_and_parse_value" and add some check for int parsing as well as bools, but doesn't matter either way to me.

I'm happy with this parsing, does make querying later a little bit easier, and it's stored as the JSON value, so the type information is actually preserved.